### PR TITLE
Internal message printer exposed as `display_dig_style`

### PR DESCRIPTION
**Pull request by tertsdiepraam**
For use in `dnsi` and `dnst` (`update` in particular). Heavily based on the dig output in `dnsi`: https://github.com/NLnetLabs/dnsi/blob/main/src/output/dig.rs. I'm not really sure about the error handling here, so feel free to scrutinize.

### CODE REVIEW

**Review by bal-e — COMMENTED** [review]
`MessagePrinter` (and the module) should be renamed to `DigPrinter`, I think.  It can also probably be placed directly inside `display_dig_style()`, but some of the code may become too nested.

It's unfortunate that printing stops the moment an error is discovered, but I don't think there's a way around that.  I'm slightly concerned that users won't notice that they're printing invalid messages -- at least some documentation is necessary on `display_dig_style()`, which mentions the behavior for messages containing parse errors.

**Comment by tertsdiepraam** [comment]
> It's unfortunate that printing stops the moment an error is discovered, but I don't think there's a way around that. I'm slightly concerned that users won't notice that they're printing invalid messages -- at least some documentation is necessary on display_dig_style(), which mentions the behavior for messages containing parse errors.

Note that this is actually not super common, because it would be when a message is incomplete and doesn't have the right number of records. If some record data is invalid then it will continue.

File: src/base/message.rs [hunk 1]
```diff
@@ -665,6 +666,13 @@ impl<Octs: Octets + ?Sized> Message<Octs> {
     }
 }
 
+/// # Printing
+impl<Octs: AsRef<[u8]>> Message<Octs> {
+    pub fn display_dig_style(&self) -> impl core::fmt::Display + '_ {
```
Comment (bal-e) [inline src/base/message.rs:671]: Small thing, but could you add a doc comment here?  It's part of the public API, after all.

Comment (tertsdiepraam) [reply src/base/message.rs:671]: Fair enough :)

**Review by bal-e — APPROVED** [review]
LGTM!

**Pull request merged**